Sources/Utility/Git.swift

```diff
@@ -12,52 +12,44 @@ import class Foundation.ProcessInfo
 import Basic
 
 extension Version {
-    static func vprefix(_ string: String) -> Version? {
-        if string.first == "v" {
-            return Version(string: String(string.dropFirst()))
-        } else {
-            return nil
+    init?(tag: String) {
+        if tag.first == "v" {
+            self.init(string: String(tag.dropFirst()))
+        }
+        else {
+            self.init(string: tag)
         }
     }
 }
 
 public class Git {
-    /// Compute the version -> tag mapping from a list of input `tags`.
-    public static func convertTagsToVersionMap(_ tags: [String]) -> [Version: String] {
+    /// Compute the version -> tags mapping from a list of input `tags`.
+    public static func convertTagsToVersionMap(_ tags: [String]) -> [Version: [String]] {
         // First, check if we need to restrict the tag set to version-specific tags.
-        var knownVersions: [Version: String] = [:]
-        for versionSpecificKey in Versioning.currentVersionSpecificKeys {
-            for tag in tags where tag.hasSuffix(versionSpecificKey) {
-                let specifier = String(tag.dropLast(versionSpecificKey.count))
-                if let version = Version(string: specifier) ?? Version.vprefix(specifier) {
-                    knownVersions[version] = tag
+        var knownVersions: [Version: [String]] = [:]
+        var versionSpecificKnownVersions: [Version: [String]] = [:]
+
+        for tag in tags {
+            for versionSpecificKey in Versioning.currentVersionSpecificKeys {
+                if tag.hasSuffix(versionSpecificKey) {
+                    let trimmedTag = String(tag.dropLast(versionSpecificKey.count))
+                    if let version = Version(tag: trimmedTag) {
+                        versionSpecificKnownVersions[version, default: []].append(tag)
+                    }
+                    break
                 }
             }
-
-            // If we found tags at this version-specific key, we are done.
-            if !knownVersions.isEmpty {
-                return knownVersions
+            
+            if let version = Version(tag: tag) {
+                knownVersions[version, default: []].append(tag)
             }
         }
-
-        // Otherwise, look for normal tags.
-        for tag in tags {
-            if let version = Version(string: tag) {
-                knownVersions[version] = tag
-            }
+        if !versionSpecificKnownVersions.isEmpty {
+            return versionSpecificKnownVersions
         }
-
-        // If we didn't find any versions, look for 'v'-prefixed ones.
-        //
-        // FIXME: We should match both styles simultaneously.
-        if knownVersions.isEmpty {
-            for tag in tags {
-                if let version = Version.vprefix(tag) {
-                    knownVersions[version] = tag
-                }
-            }
+        else {
```

Comment:
Nitpick: can you put else on the same line as the if closing brace to follow the project's informal style?